### PR TITLE
switch benchmarking environment from Python 3.6 to Python 3.8

### LINKED ISSUE #11: update benchmark env to > 3.7
The benchmarks for 2.0 and greater fail because we discontinued 3.6 support:
```
· Getting existing hashes
· Creating environments
· Discovering benchmarks
·· Uninstalling from conda-py3.6-Cython-mdanalysistests-numpy-scipy
·· Building 39850215 <develop> for conda-py3.6-Cython-mdanalysistests-numpy-scipy
·· Error running ~/MDA/repositories/mdanalysis/benchmarks/env/80c5ba7efaa89366b97da3ced85c0c44/bin/python setup.py build (exit status 255)
  STDOUT -------->
  MDAnalysis requires Python 3.7 or better. Python 3.6 detected
  Please upgrade your version of Python.
  STDERR -------->
```

Update the benchmarking environment to 3.8.

### PR DESCRIPTION
**Pull request by orbeckst**
fix #11

### CODE REVIEW

**Comment by orbeckst** [comment]
I am currently running this manually. Will report back if it's working.

Comments welcome @jbarnoud @IAlibay

**Comment by orbeckst** [comment]
This might take a while...
```
+ asv run -e --config ..../mdanalysis/benchmarks/asv_c3potato.conf.json NEW
· Getting existing hashes
· Creating environments...............................
· Discovering benchmarks..
·· Uninstalling from conda-py3.8-Cython-mdanalysistests-numpy-scipy
·· Building 3c0875c7 <develop> for conda-py3.8-Cython-mdanalysistests-numpy-scipy................
·· Installing 3c0875c7 <develop> into conda-py3.8-Cython-mdanalysistests-numpy-scipy.
· Running 1701 total benchmarks (21 commits * 1 environments * 81 benchmarks)
[  0.00%] · For mdanalysis commit 3c0875c7 <develop>:
.....
```

File: results/benchmarks.json [hunk 1]
```diff
@@ -1147,6 +1147,94 @@
         "version": "e17b327697d267cf044412cc544be3b2e309edad29b10d0ab5af64a7b847d269",
         "warmup_time": -1
     },
+    "ag_methods.CompoundSplitting.time_center_compounds": {
```
Comment (orbeckst) [inline results/benchmarks.json]: Ignore these changes... they should have been on master but I added them to the wrong branch.

**Comment by jbarnoud** [comment]
It would make sense to make the change in `asv.conf.json` as well to avoid having that file lagging behind.

**Comment by jbarnoud** [comment]
Also, it looks like the change worked. @orbeckst can you confirm?

**Comment by orbeckst** [comment]
It ran. The script stopped after 20h (because I forgot that there's a timeout in the cron job script) but it will keep working on the missing commits every night. I re-enabled the cron job so benchmarks should be slowly updating. 

The line that runs the benchmarks https://github.com/MDAnalysis/benchmarks/blob/32daf276019cec212e164a58f6ddc0f2d900ddc3/scripts/cron.sh#L67 is currently going back to the 0.17.0 merge commit. Some of the tests seem to exercise code that is not present any more (such as setting core.flags in `GeoSelectionBench.time_geometric_selections`). The other tests still run but we should revisit the tests and then change how far back we want to go (probably the 1.0.0 merge is good enough).

**Comment by orbeckst** [comment]
@jbarnoud I updated the template asv.config.json.

Anything else that should be done or can I merge and then we can raise issues for test updating?

**Comment by orbeckst** [comment]
@IAlibay in https://github.com/MDAnalysis/benchmarks/issues/11#issuecomment-1005299079 wants me to change the merge commit start to 1.0.1 so I'll do that right away,

**Comment by orbeckst** [comment]
Having issues to get the "run missing commits" (history) to work again (regardless of release-0.17.0 or release-1.0.1)
```
asv run -e -j 4 --config .../benchmarking/benchmarks/config/asv_c3potato.conf.json  "release-0.17.0..HEAD --merges" --skip-existing
· Error running .../miniconda3/envs/benchmark/bin/git rev-list --first-parent release-0.17.0..HEAD --merges (exit status 128)
  STDOUT -------->

  STDERR -------->
  fatal: ambiguous argument 'release-0.17.0..HEAD': unknown revision or path not in the working tree.
  Use '--' to separate paths from revisions, like this:
  'git <command> [<revision>...] -- [<file>...]'
```
When running from ASV_RUNDIR
```
.../miniconda3/envs/benchmark/bin/git rev-list --first-parent release-0.17.0..HEAD --merges
```
I get ~140 commits.

Ideas??

**Comment by jbarnoud** [comment]
The version tags aren't on the same branch. Could it be it? I know it did bite me with bisects. On 7 Feb 2022 17:47, Oliver Beckstein ***@***.***> wrote:

Having issues to get the "run missing commits" (history) to work again (regardless of release-0.17.0 or release-1.0

asv run -e -j 4 --config .../benchmarking/benchmarks/config/asv_c3potato.conf.json  "release-0.17.0..HEAD --merges" --skip-existing

· Error running .../miniconda3/envs/benchmark/bin/git rev-list --first-parent release-0.17.0..HEAD --merges (exit status 128)

  STDOUT -------->



  STDERR -------->

  fatal: ambiguous argument 'release-0.17.0..HEAD': unknown revision or path not in the working tree.

  Use '--' to separate paths from revisions, like this:

  'git <command> [<revision>...] -- [<file>...]'



When running from ASV_RUNDIR

.../miniconda3/envs/benchmark/bin/git rev-list --first-parent release-0.17.0..HEAD --merges



I get ~140 commits.

Ideas??

**Comment by orbeckst** [comment]
I don't think it depends on which branch I am. The `git rev-list` works regardless if I am on master or develop.

Trying out a few more things, especially if there's a difference between the cron script env and commandline.

**Comment by orbeckst** [comment]
Fine... it works in the cron env:
```
+ timeout 96h asv run -e -j 4 --config .../repositories/mdanalysis/benchmarks/asv_c3potato.conf.json 'release-1.0.1..HEAD --merges' --skip-existing
· Creating environments
· Discovering benchmarks
·· Uninstalling from conda-py3.8-Cython-mdanalysistests-numpy-scipy
·· Building 5bf6359a <develop~21> for conda-py3.8-Cython-mdanalysistests-numpy-scipy................
·· Installing 5bf6359a <develop~21> into conda-py3.8-Cython-mdanalysistests-numpy-scipy.
· Running 2511 total benchmarks (31 commits * 1 environments * 81 benchmarks)
[ 83.87%] · For mdanalysis commit ca2cbe40 <release-2.0.0-beta~143>:
[ 83.87%] ·· Building for conda-py3.8-Cython-mdanalysistests-numpy-scipy.
[ 83.87%] ·· Benchmarking conda-py3.8-Cython-mdanalysistests-numpy-scipy
[ 83.87%] ··· Importing benchmark suite produced output:
...
```

I merged locally into master for testing so please _don't merge this PR from GitHub_. 

I am now manually catching up on history.

**Pull request merged**